apiv2/db_import/common/config.py

```diff
@@ -0,0 +1,103 @@
+import json
+import logging
+from datetime import datetime
+from pathlib import PurePath
+from typing import TYPE_CHECKING, Any
+
+from botocore.exceptions import ClientError
+from sqlalchemy.orm import Session
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+else:
+    S3Client = object
+
+logger = logging.getLogger("config")
+
+
+class DBImportConfig:
+    s3_client: S3Client
+    bucket_name: str
+    s3_prefix: str
+    https_prefix: str
+    session: Session
+
+    def __init__(
+        self,
+        s3_client: S3Client,
+        bucket_name: str,
+        https_prefix: str,
+        session: Session,
+    ):
+        self.s3_client = s3_client
+        self.bucket_name = bucket_name
+        self.s3_prefix = f"s3://{bucket_name}"
+        self.https_prefix = https_prefix if https_prefix else "https://files.cryoetdataportal.cziscience.com"
+        self.session = session
+
+    def get_db_session(self) -> Session:
+        return self.session
+
+    def find_subdirs_with_files(self, prefix: str, target_filename: str) -> list[str]:
+        paginator = self.s3_client.get_paginator("list_objects_v2")
+        logger.info("looking for prefix %s", prefix)
+        pages = paginator.paginate(Bucket=self.bucket_name, Prefix=prefix, Delimiter="/")
+        result = []
+        for page in pages:
+            for obj in page.get("CommonPrefixes", []):
+                try:
+                    subdir = obj["Prefix"]
+                    self.s3_client.head_object(Bucket=self.bucket_name, Key=f"{subdir}{target_filename}")
+                    result.append(subdir)
+                except Exception:
+                    continue
+        return result
+
+    def glob_s3(self, prefix: str, glob_string: str, is_file: bool = True):
+        paginator = self.s3_client.get_paginator("list_objects_v2")
+        if prefix.startswith("s3://"):
+            prefix = "/".join(prefix.split("/")[3:])
```

Comment:
I added these two lines to this method to make it a little less fussy to work with.